src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -101,20 +101,56 @@ export const StudyProposal: React.FC<{ memberSlug: string }> = ({ memberSlug })
             })
             router.push(`/researcher/dashboard`)
         },
-        onError: async (error, _, context: { studyId: string; studyJobId: string } | undefined) => {
-            console.error(error)
+        onError: async (error: unknown, _variables, context: { studyId: string; studyJobId: string } | undefined) => {
+            console.error('Study proposal submission error:', error)
+
+            let title = 'Submission Failed' // Default title
+            let message = 'An unexpected error occurred.' // Default message
+            let needsCleanup = false
+            let rawErrorMessage = error instanceof Error ? error.message : 'Unknown error details'; // Capture raw message
+
+            // Check for specific access denied error
+            if (error instanceof Error && error.message.includes('You are not a member of this organization')) {
+                title = 'Access Denied'
+                message = 'You can only submit study proposals to organizations you are a member of.'
+                needsCleanup = false
+            } else if (context) {
+                // Use the requested title for upload/processing errors
+                title = 'Failed to upload file'
```

Comment:
the `uploadFile` method already throws an error, we can probably just display those errors with a notification.show on line 30:
```
    if (!response.ok) {
        throw new Error(`failed to upload file ${await response.text()}`)
    }
```